### ISSUE TITLE
[Bug] Calculating transform color

### ISSUE THREAD

**Issue by mgamperl** [dschu012/d2s#29]
There seems to be a bug when calculating the transform color for the magical_name_ids:
```typescript
//line 115 in attribute_enhancer.ts
for (let i = 0; i < 6; i++) {
  let id = item.magical_name_ids[i];
  if (id) {
    if (i % 5 == 0 && constants.magic_suffixes[id].tc) {
      item.transform_color = constants.magic_suffixes[id].tc;
    } else if (constants.magic_prefixes[id].tc) {
      item.transform_color = constants.magic_prefixes[id].tc;
    }
  }
}
```
from what I have read about the file format in various sources the 6 different ids mean the following:
```
0: first prefix
1: first suffix
2: second prefix
3: second suffix
4: third prefix
5: third suffix
```
so the logic should be something like this:
```typescript
for(let i = 0; i < 6; i++) {
  let id = item.magical_name_ids[i];
  if (id) {
    if (i % 2 == 0 && constants.magic_prefixes[id] && constants.magic_prefixes[id].tc) { // even is prefixes
      item.transform_color = constants.magic_prefixes[id].tc;            
    } else if(constants.magic_suffixes[id] && constants.magic_suffixes[id].tc) { // odd is suffixes
      item.transform_color = constants.magic_suffixes[id].tc;
    }            
  }
}
```
I am afraid that I miss something but I can not explain why the `i %5 == 0` is used in the current implementation and I can not even get my head around of what the current implementation tries to achieve.

I just discovered that problem because I use a D2 version modded with Project Diablo 2 and there it seems that the id's of prefixes and suffixes differ so it will crash with trying to read `.tc from undefined`. I think under "normal" circumstances his will not throw an error. 

As I said maybe I'm missing something but I think if this is a bug it will not be easily discovered as such because under normal conditions this color transformations seems to be very rarely used in that context.

**Comment by dschu012** [comment]
Lol, I'm not quite sure why it is `i%5==0` either. Wrote it a long time ago, not sure what I was thinking when I wrote that line.

**Comment by mgamperl** [comment]
Will be fixed with #31
